File: Number Theory/2523.Closest Prime Numbers in Range/Solution.py
class Solution:
    def closestPrimes(self, left: int, right: int) -> List[int]:
        primes = [False] * (right + 1)
        for i in range(2, right + 1):
            if not primes[i]:
                for j in range(i * i, right + 1, i):
                    primes[j] = True

        ans = None
        last = -1
        for i in range(max(left, 2), right + 1):
            if not primes[i]:
                if last != -1 and (not ans or ans[1] - ans[0] > i - last):
                    ans = [last, i]
                last = i
        return [-1, -1] if not ans else ans

File: Number Theory/2523.Closest Prime Numbers in Range/Solution.java
class Solution {
    public int[] closestPrimes(int left, int right) {
        boolean[] primes = new boolean[right + 1];
        for (long i = 2; i <= right; i++) {
            if (!primes[(int)i]) {
                for (long j = i * i; j <= right; j += i) {
                    primes[(int)j] = true;
                }
            }
        }
        int[] ans = null;
        int last = -1;
        for (int i = Math.max(left, 2); i <= right; i++) {
            if (!primes[i]) {
                if (last != -1 && (ans == null || ans[1] - ans[0] > i - last)) {
                    ans = new int[]{last, i};
                }
                last = i;
            }
        }
        return ans == null ? new int[]{-1, -1,} : ans;
    }
}